routing/probability_estimator_test.go

```diff
@@ -180,3 +207,70 @@ func TestProbabilityEstimatorMix(t *testing.T) {
 		testTime, node2, 100, testCapacity, expectedNodeProb*0.75,
 	)
 }
+
+// TestCapacityCutoff tests the mathematical expression and limits for the
+// capacity factor.
+func TestCapacityCutoff(t *testing.T) {
+	t.Parallel()
+
+	capacitySat := 1_000_000
+	capacityMSat := capacitySat * 1000
+
+	tests := []struct {
+		name           string
+		amountMsat     int
+		expectedFactor float64
+	}{
+		{
+			name:           "zero amount",
+			expectedFactor: 1,
+		},
+		{
+			name:           "low amount",
+			amountMsat:     capacityMSat / 10,
+			expectedFactor: 0.998,
+		},
+		{
+			name:           "half amount",
+			amountMsat:     capacityMSat / 2,
+			expectedFactor: 0.924,
+		},
+		{
+			name: "cutoff amount",
+			amountMsat: int(
+				capacityCutoffFraction * float64(capacityMSat),
+			),
+			expectedFactor: 0.5,
+		},
+		{
+			name:           "high amount",
+			amountMsat:     capacityMSat * 80 / 100,
+			expectedFactor: 0.377,
+		},
+		{
+			// Even when we spend the full capacity, we still want
+			// to have some residual to not throw away routes due to
+			// minimal probability constraints.
```

Comment:
what is this "minimal probability constraints"?

Reply:
I updated the comment and explained this a bit more in the comment about the parameters. Currently, we have a requirement for a minimal probability for routes. If we dip below that we'd not consider the route.